src/numbers/barrett.rs

```diff
@@ -0,0 +1,51 @@
+//! # Barrett Division and Remainder
+
+/// See <https://github.com/kth-competitive-programming/kactl/blob/main/content/various/FastMod.h>
+///
+/// # Example
+/// ```
+/// use programming_team_code_rust::numbers::barrett::Barrett;
+/// let barr = Barrett::new(6_u32);
+/// assert_eq!(barr.div(20_u64), (3, 2));
+/// ```
+///
+/// # Panics
+/// ```panic
+/// use programming_team_code_rust::numbers::barrett::Barrett;
+/// let barr = Barrett::new(0_u32);
+/// ```
+pub struct Barrett {
+    /// denominator
+    pub denom: u64,
+    im: u64,
+}
+
+impl Barrett {
+    /// Creates a new Barrett struct
+    ///
+    /// # Complexity
+    /// - Time: O(1)
+    /// - Space: O(1)
+    pub fn new(denom: u32) -> Self {
```

Comment:
not sure how I feel about this; but basically this struct only works for `denom < u32::MAX/2` 

but I want to use `u64` for modulo operations to not have to cast so often